File: Recursion/lecture 7/Fraz_recursion_7_1.java
ArrayList<ArrayList<Integer>> ans;
ArrayList<Integer> temp;

    ArrayList<ArrayList<Integer>> helper(ArrayList<Integer> v, int i) {
        ArrayList<ArrayList<Integer>> partialAns = helper(v, i + 1);

        for (ArrayList<Integer> x : partialAns) {
            ans.add(x);
        }
        for (ArrayList<Integer> x : partialAns) {

            temp.add(v.get(i));
            for (int y : x) {
                temp.add(y);
            }
            ans.add(temp);
        }
        return ans;
    }

    ArrayList<ArrayList<Integer>> pwset(ArrayList<Integer> v) {
        return helper(v, 0);
    }

File: Recursion/lecture 7/Fraz_recursion_7_1.cpp
vector<vector<int>> helper(vector<int> &v,int i)
{
    //base condition
    if(i>=v.size()) return {{}};
    vector<vector<int>> partialAns = helper(v,i+1);
    vector<vector<int>> ans;
    for(vector<int> x:partialAns)
    {
        ans.push_back(x);
    }
    for(vector<int> x:partialAns){
        vector<int> temp;
        temp.push_back(v[i]);
        for(int y:x)
        {
            temp.push_back(y);
        }
        ans.push_back(temp);
    }
    return ans;
}

vector<vector<int>> pwset(vector<int>v)
{
    return helper(v,0);
}
